### PR TITLE
PRJ-770: Update GotIt message position if widget is moved

### PR DESCRIPTION
**Pull request by IKrukov-HORIS**
Use PositionTracker in showMessage function to track widget position

### CODE REVIEW

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt [hunk 1]
```diff
@@ -41,7 +43,18 @@ internal fun showMessage(project: Project, header: String, message: String): Boo
...
   if (widget is ProjectorStatusWidget) {
     val gotItMessage = GotItMessage.createMessage(header, message).setDisposable(widget)
-    gotItMessage.show(RelativePoint.getCenterOf(widget.component), Balloon.Position.above)
+
+    with(widget.component) {
+      gotItMessage.show(
+        object : PositionTracker<Balloon?>(this) {
+          override fun recalculateLocation(baloon: Balloon): RelativePoint {
+            return RelativePoint(widget.component, Point(this@with.size.width / 2,
+                                                         this@with.size.height / 2))
```
Comment (SerVB) [inline projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt:52]: I just spotted this PR. I haven't tested but I'm wondering if this code can be improved. For example:

```suggestion
            return RelativePoint(this@with, Point(this@with.size.width / 2,
                                                         this@with.size.height / 2))
```

Or even (as before):

```suggestion
            return RelativePoint.getCenterOf(this@with)
```

Comment (IKrukov-HORIS) [reply projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt]: Nice catch! I like more your last suggestion. Commit it with one minor change.

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt [hunk 2]
```diff
@@ -41,7 +42,17 @@ internal fun showMessage(project: Project, header: String, message: String): Boo
...
+
+    with(widget.component) {
+      gotItMessage.show(
+        object : PositionTracker<Balloon?>(this) {
+          override fun recalculateLocation(baloon: Balloon): RelativePoint {
+            return RelativePoint.getCenterOf(this@with)
+          }
+        },
+        Balloon.Position.above)
+    }
```
Comment (SerVB) [inline projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt:54]: Minor: looks like now we can improve it even more ;)

```suggestion
    gotItMessage.show(
      object : PositionTracker<Balloon?>(widget.component) {
        override fun recalculateLocation(baloon: Balloon) = RelativePoint.getCenterOf(widget.component)
      },
      Balloon.Position.above,
    )
```

@ARTI1208, please take a look, test it, and take the final decision. I'm going out of this PR.

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt [hunk 3]
```diff
@@ -41,7 +42,14 @@ internal fun showMessage(project: Project, header: String, message: String): Boo
 
   if (widget is ProjectorStatusWidget) {
     val gotItMessage = GotItMessage.createMessage(header, message).setDisposable(widget)
-    gotItMessage.show(RelativePoint.getCenterOf(widget.component), Balloon.Position.above)
+
+    gotItMessage.show(
+      object : PositionTracker<Balloon?>(widget.component) {
```
Comment (ARTI1208) [inline projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt:47]: Is nullability needed here?

**Review by ARTI1208 — APPROVED** [review]

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt (hunk 3 above)
Comment (IKrukov-HORIS) [reply projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/Messages.kt]: No, this is unnecessary. Fixed.

**Comment by ARTI1208** [comment]
Please, add this change to changelog

**Pull request merged**